src/ucp/core/ucp_ep.c

```diff
@@ -849,6 +850,12 @@ int ucp_ep_config_is_equal(const ucp_ep_config_key_t *key1,
         }
     }
 
+    for (i = 0; i < ucs_popcount(key1->reachable_md_map); ++i) {
```

Comment:
why do we need this check, if  dst md index is checked for all lanes?
what is the use case when we have identical lanes, but different mds?

Reply:
it's possible in theory that some of the remote MDs are not currently reachable by existing lanes, so we didn't check them in the first loop. dst_md_cmpts should cover all reachable destination mds, as deduced from the worker address supplied initially. to support ucp_ep transport switch in the future ,